agent/rpc/middleware/rate_limit_mappings.go

```diff
@@ -0,0 +1,144 @@
+
+	"ConnectCA.ConfigurationGet": rate.OperationTypeRead,
+	"ConnectCA.ConfigurationSet": rate.OperationTypeWrite,
+	"ConnectCA.Roots":            rate.OperationTypeRead,
+	"ConnectCA.Sign":             rate.OperationTypeWrite,
+	"ConnectCA.SignIntermediate": rate.OperationTypeWrite,
+
+	"Coordinate.ListDatacenters": rate.OperationTypeRead,
+	"Coordinate.ListNodes":       rate.OperationTypeRead,
+	"Coordinate.Node":            rate.OperationTypeRead,
+	"Coordinate.Update":          rate.OperationTypeWrite,
+
+	"DiscoveryChain.Get": rate.OperationTypeRead,
+
+	"FederationState.Apply":            rate.OperationTypeWrite,
+	"FederationState.Delete":           rate.OperationTypeWrite,
+	"FederationState.Get":              rate.OperationTypeRead,
+	"FederationState.List":             rate.OperationTypeRead,
+	"FederationState.ListMeshGateways": rate.OperationTypeRead,
+
+	"Health.ChecksInState": rate.OperationTypeRead,
+	"Health.NodeChecks":    rate.OperationTypeRead,
+	"Health.ServiceChecks": rate.OperationTypeRead,
+	"Health.ServiceNodes":  rate.OperationTypeRead,
+
+	"Intention.Apply": rate.OperationTypeWrite,
+	"Intention.Check": rate.OperationTypeRead,
+	"Intention.Get":   rate.OperationTypeRead,
+	"Intention.List":  rate.OperationTypeRead,
+	"Intention.Match": rate.OperationTypeRead,
+
+	"Internal.CatalogOverview":               rate.OperationTypeRead,
+	"Internal.EventFire":                     rate.OperationTypeWrite,
+	"Internal.ExportedPeeredServices":        rate.OperationTypeRead,
+	"Internal.ExportedServicesForPeer":       rate.OperationTypeRead,
+	"Internal.GatewayIntentions":             rate.OperationTypeRead,
+	"Internal.GatewayServiceDump":            rate.OperationTypeRead,
+	"Internal.IntentionUpstreams":            rate.OperationTypeRead,
+	"Internal.IntentionUpstreamsDestination": rate.OperationTypeRead,
+	"Internal.KeyringOperation":              rate.OperationTypeRead,
+	"Internal.NodeDump":                      rate.OperationTypeRead,
+	"Internal.NodeInfo":                      rate.OperationTypeRead,
+	"Internal.PeeredUpstreams":               rate.OperationTypeRead,
+	"Internal.ServiceDump":                   rate.OperationTypeRead,
+	"Internal.ServiceGateways":               rate.OperationTypeRead,
+	"Internal.ServiceTopology":               rate.OperationTypeRead,
+
+	"KVS.Apply":    rate.OperationTypeWrite,
+	"KVS.Get":      rate.OperationTypeRead,
+	"KVS.List":     rate.OperationTypeRead,
+	"KVS.ListKeys": rate.OperationTypeRead,
+
+	"Operator.AutopilotGetConfiguration": rate.OperationTypeRead,
+	"Operator.AutopilotSetConfiguration": rate.OperationTypeWrite,
+	"Operator.AutopilotState":            rate.OperationTypeRead,
+	"Operator.RaftGetConfiguration":      rate.OperationTypeRead,
+	"Operator.RaftRemovePeerByAddress":   rate.OperationTypeWrite,
+	"Operator.RaftRemovePeerByID":        rate.OperationTypeWrite,
+	"Operator.ServerHealth":              rate.OperationTypeRead,
```

Comment:
I commented about this in the grpc [PR](https://github.com/hashicorp/consul/pull/15564).

I think we need to add a third type, (eg: bypass). All operator endpoints should not be rate limited as they can be used by an operator in an outage situation.

Reply:
Rather than assign additional meaning to the `OperationType` which seems to be doing a good job of capturing read/write intent, how about capturing "exempt from rate limiting" status separately?
```Go
// endpoints that are exempt from rate limiting
var rpcExemptOperations = []string {
    "Operator.AutopilotState",
    ...
}
```

Code that will probably end up depending on `OperationType` to have a read/write value will be more straightforward and not have to be concerned about non-read/write contexts. e.g. emitting metrics with a read/write label.